app/grandchallenge/reader_studies/models.py

```diff
@@ -123,6 +135,38 @@
 as for each case are displayed in the ``statistics`` view.
 """
 
+
+#: Supported image ports.
+IMAGE_PORTS = [
+    "main",
+    "secondary",
+    "tertiary",
+    "quaternary",
+    "quinary",
+    "senary",
+    "septenary",
+    "octonary",
+    "nonary",
+    "denary",
+]
+
+
+def generate_image_port_properties(image_ports):
+    properties = {}
+    image_port_overlays = [f"{port}-overlay" for port in image_ports]
+    for port in image_ports + image_port_overlays:
+        properties[port] = {
+            "$id": f"#/items/properties/{port}",
+            "type": "string",
+            "title": f"The {port[0].upper() + port[1:]} Schema",
+            "default": "",
+            "examples": [f"im_{port}.mhd"],
+            "pattern": "^(.*)$",
+        }
+    return properties
+
```

Comment:
Not sure what the cleanest way of generating this is. Having them in hardcoded (20 times) seemed a bit much.

Reply:
You may have been able to do it in a single generator but it's a good approach.

Reply:
Sorry @jmsmkn, not sure that I follow. So just to check, do mean an in-line generator instead of a rogue function?